### PR TITLE
add SLIP encoding

### PR DESCRIPTION
**Pull request by avilleret**

### CODE REVIEW

File: src/PacketSerial.h [hunk 1]
```diff
@@ -146,3 +147,4 @@ class PacketSerial_
 
 
 typedef PacketSerial_<COBS> PacketSerial;
+typedef PacketSerial_<SLIP, SLIP::END, 8192> SLIPPacketSerial;
```
Comment (bakercp) [inline src/PacketSerial.h]: I'm wondering if the default buffer should be this big on an embedded system.  Perhaps leave it default -- something like:

```c++
 typedef PacketSerial_<COBS> PacketSerial;
 typedef PacketSerial_<COBS> COBSPacketSerial;
 typedef PacketSerial_<SLIP, SLIP::END> SLIPPacketSerial;
```

Comment (bakercp) [inline src/PacketSerial.h]: Also, added a similar type def for cobs ...

**Comment by bakercp** [comment]
Also, while you're in there would you mind updating the readme to mention that SLIP is an option as well?

Thanks!!

**Comment by avilleret** [comment]
I made some changes, take your time to review them before merging (or not :-) )

**Comment by avilleret** [comment]
ping

**Comment by bakercp** [comment]
Thanks!

**Pull request merged**